### PR TITLE
Skip collection exists check when creating collections and folders

### LINKED ISSUE #23987: FORBIDDEN permission error when creating a folder
### Describe the Bug

It seems that the folder's permissions is missing since version `11.1.2` (which is tested). Please see steps to reproduce. The issue is observed with both PostgreSQL and SQLite databases.

### To Reproduce

Start a fresh instance of Directus:

```
docker run -it --rm -p 8055:8055 directus/directus:11.1.2
```

Then open the browser and:

1. Login with the generated admin account
2. Add a new folder under Settings / Data Model
3. Upon saving, it renders the error: `[FORBIDDEN] You don't have permission to "read" from collection "<Folder Name>" or it does not exist.`

### Directus Version

v11.1.2

### Hosting Strategy

Self-Hosted (Docker Image)

### Database

SQLite/PostgreSQL 15

### PR DESCRIPTION
**Pull request by licitdev**
## Scope

What's changed:

- Skipped collection exists check when creating collections and folders
- Removed unnecessary schema refreshing from #23888

## Potential Risks / Drawbacks

- Lorem ipsum dolor sit amet
- Consectetur adipiscing elit

## Review Notes / Questions

- I would like to lorem ipsum
- Special attention should be paid to dolor sit amet

---

Fixes #23987
Closes SER-409

### CODE REVIEW

File: api/src/permissions/modules/validate-access/validate-access.ts [hunk 1]
```diff
@@ -17,9 +17,13 @@ export interface ValidateAccessOptions {
  * optional primary keys. This is done by reading the item from the database using the access
  * control rules and checking if we got the expected result back
  */
-export async function validateAccess(options: ValidateAccessOptions, context: Context) {
+export async function validateAccess(
+	options: ValidateAccessOptions,
+	context: Context,
+	skipCollectionExistsCheck = false,
```
Comment (hanneskuettner) [inline api/src/permissions/modules/validate-access/validate-access.ts:23]: Can we add this as an optional option in the `ValidateAcessOptions`? Seems like that is the place where it should live IMO. Easier to understand than a positional parameter.

Comment (licitdev) [reply api/src/permissions/modules/validate-access/validate-access.ts]: Absolutely! 💯 Refactored in https://github.com/directus/directus/pull/23991/commits/c52206b04dd4d306162b354f7ed86b65ed2ffd36

File: api/src/services/collections.ts [hunk 2]
```diff
@@ -204,12 +204,11 @@ export class CollectionsService {
...
 
-			// Refresh the schema for subsequent reads
-			this.schema = await getSchema();
-
 			if (opts?.emitEvents !== false && nestedActionEvents.length > 0) {
+				const updatedSchema = await getSchema();
+
 				for (const nestedActionEvent of nestedActionEvents) {
-					nestedActionEvent.context.schema = this.schema;
+					nestedActionEvent.context.schema = updatedSchema;
```
Comment (hanneskuettner) [inline api/src/services/collections.ts:211]: We should probably revert both updates of the schema that were implemented in https://github.com/directus/directus/pull/23888/files and add the skip flag in the second place as well.

Comment (licitdev) [reply api/src/services/collections.ts:211]: Ahhhh yes there's another one too. Thanks!

Comment (licitdev) [reply api/src/services/collections.ts:211]: The skip flag is only added in `readMany`.

Comment (licitdev) [reply api/src/services/collections.ts:211]: Refactored in https://github.com/directus/directus/pull/23991/commits/4773da9c0d054ca8e4327d19d883861be82c00c6

Comment (hanneskuettner) [reply api/src/services/collections.ts:211]: Oh yeah, `validateAccess` is only called in `readMany`

**Review by hanneskuettner — APPROVED** [review]
Great fix! LGTM 🎸

**Pull request merged**